File: execute.mongo.js
// https://www.codementor.io/codementorteam/how-to-use-json-files-in-node-js-85hndqt32
// https://nodejs.org/docs/latest/api/process.html#process_process_argv
//var format = require('./node_format')
var J = require('./'+process.argv[2]);
var d = process.argv[3];
var t = process.argv[4];
var keyword = 'filepath';
var JQ = [];
// https://coderwall.com/p/_kakfa/javascript-iterate-through-object-keys-and-values
for (var m in J) {
  K = J[m];
  for (var n in K) {
    dict = K[n]
    if( Object.keys(dict).length>1 ){
      dict[keyword] = m;
      js = JSON.stringify(dict);
      JQ.push( `db.${t}.insert(${js})` );
    }
  }
}

console.log( `use ${d}` );
console.log( 'db.dropDatabase()' );
console.log( `use ${d}` );

console.log( JQ.join("\n") );

console.log('show dbs')

File: execute.mongo.py
#!/usr/bin/python3
import json,sys;
t = 'id3v2'
keyword = 'filepath'
with open(sys.argv[1],encoding='utf8') as mongo:
  JQ = []
  J = json.load(mongo)
  for j in J:
    u = J[j]
    if len(u)>1 : 
      u.update({keyword:j});
      js = json.dumps(u)
      JQ.append( 'db.{}.insert({})'.format(t,js) )

print("""
db.dropDatabase()
use zin
""")
print( "\n".join(JQ) )

#https://stackoverflow.com/questions/5466451/how-can-i-print-literal-curly-brace-characters-in-python-string-and-also-use-fo

print( 'db.{}.find( {{"{}":{{ {}:{} }}}} ).limit(1)'.format(t,'duration','$gt',60*5) )

print( 'db.{}.aggregate([  {{"$group": {{_id:"${}",count:{{$sum:1}}}}}}  ])'.format(t,'TALB') )

#print('show dbs')

